test/unit/org/apache/cassandra/service/StorageServiceServerTest.java

```diff
@@ -658,4 +660,26 @@ public void isReplacingSameHostAddressAndHostIdTest() throws UnknownHostExceptio
             Assert.assertFalse(StorageService.instance.isReplacingSameHostAddressAndHostId(differentHostId));
         }
     }
-}
\ No newline at end of file
+
+    @Test
+    public void testIsDecommissionNotFailed()
+    {
+        assertFalse(StorageService.instance.isDecommissionFailed());
+        assertEquals(0, StorageMetrics.errorDecommissiong.getCount());
+    }
+
+    @Test
```

Comment:
I think we can also add a new test case use ToolRunner.invokeNodetool see GossipInfoTest
we can use at frist set the node's mode to Un-normal and then use ToolRunner.invokeNodetool  to get the status of descommission.